src/Microsoft.PowerShell.ConsoleHost/host/msh/UpdatesNotification.cs

```diff
@@ -20,24 +20,41 @@ namespace Microsoft.PowerShell
     /// </summary>
     internal static class UpdatesNotification
     {
-        private const string UpdateCheckOptOutEnvVar = "POWERSHELL_UPDATECHECK_OPTOUT";
-        private const string Last3ReleasesUri = "https://api.github.com/repos/PowerShell/PowerShell/releases?per_page=3";
-        private const string LatestReleaseUri = "https://api.github.com/repos/PowerShell/PowerShell/releases/latest";
+        private const string UpdateCheckEnvVar = "POWERSHELL_UPDATECHECK";
 
-        private const string SentinelFileName = "_sentinel_";
-        private const string DoneFileNameTemplate = "sentinel-{0}-{1}-{2}.done";
-        private const string DoneFileNamePattern = "sentinel-*.done";
-        private const string UpdateFileNameTemplate = "update_{0}_{1}";
-        private const string UpdateFileNamePattern = "update_v*.*.*_????-??-??";
-
-        private static readonly EnumerationOptions s_enumOptions = new EnumerationOptions();
-        private static readonly string s_cacheDirectory = Path.Combine(Platform.CacheDirectory, PSVersionInfo.GitCommitId);
+        private static readonly string s_cacheDirectory;
+        private static readonly string s_sentinelFileName;
+        private static readonly string s_doneFileNameTemplate;
+        private static readonly string s_doneFileNamePattern;
+        private static readonly string s_updateFileNameTemplate;
+        private static readonly string s_updateFileNamePattern;
+        private static readonly EnumerationOptions s_enumOptions;
+        private static readonly NotificationType s_notificationType;
```

Comment:
What is benefits static vs const here?

Reply:
They cannot be made const because the values are constructed based on the current notification type.